### PR TITLE
fix pg controller create redundancy podGroup when schedulerName isn't matched

### PR DESCRIPTION
**Pull request by liuyuanchun11**
fix issue #3671 

Check whether the value of pod.spec.schedulerName is matched before creating a podGroup.

### CODE REVIEW

**Review by hwdef — COMMENTED** [review]
/lgtm

File: pkg/controllers/podgroup/pg_controller_handler.go [hunk 1]
```diff
@@ -93,6 +94,10 @@ func (pg *pgcontroller) addReplicaSet(obj interface{}) {
 		if podList != nil && len(podList.Items) > 0 {
 			pod := podList.Items[0]
 			klog.V(4).Infof("Try to create podgroup for pod %s/%s", pod.Namespace, pod.Name)
+			if !slices.Contains(pg.schedulerNames, pod.Spec.SchedulerName) {
+				klog.V(4).Infof("pod %s/%s field SchedulerName is not matched", pod.Namespace, pod.Name)
```
Comment (Monokaix) [inline pkg/controllers/podgroup/pg_controller_handler.go:98]: Please change pod to upper case.

Comment (Monokaix) [reply pkg/controllers/podgroup/pg_controller_handler.go:98]: And we can use `klog.Kobj` to print pod.namespace and pos.name.

Comment (liuyuanchun11) [reply pkg/controllers/podgroup/pg_controller_handler.go]: done

File: pkg/controllers/podgroup/pg_controller_handler.go [hunk 2]
```diff
@@ -87,15 +88,19 @@ func (pg *pgcontroller) addReplicaSet(obj interface{}) {
...
 			metav1.ListOptions{LabelSelector: metav1.FormatLabelSelector(&selector)})
 		if err != nil {
-			klog.Errorf("Failed to list pods for ReplicaSet <%s/%s>: %v", rs.Namespace, rs.Name, err)
+			klog.Errorf("Failed to list pods for ReplicaSet %s: %v", klog.KObj(rs), err)
 			return
 		}
 		if podList != nil && len(podList.Items) > 0 {
 			pod := podList.Items[0]
-			klog.V(4).Infof("Try to create podgroup for pod %s/%s", pod.Namespace, pod.Name)
+			klog.V(4).Infof("Try to create podgroup for pod %s", klog.KObj(&pod))
```
Comment (Monokaix) [inline pkg/controllers/podgroup/pg_controller_handler.go:96]: klog.KObj(pod)) is ok: )

File: pkg/controllers/podgroup/pg_controller_handler.go [hunk 3]
```diff
@@ -87,15 +88,19 @@ func (pg *pgcontroller) addReplicaSet(obj interface{}) {
...
+			if !slices.Contains(pg.schedulerNames, pod.Spec.SchedulerName) {
+				klog.V(4).Infof("Pod %s field SchedulerName is not matched", klog.KObj(&pod))
+				return
+			}
 			err := pg.createNormalPodPGIfNotExist(&pod)
 			if err != nil {
-				klog.Errorf("Failed to create PodGroup for pod <%s/%s>: %v", pod.Namespace, pod.Name, err)
+				klog.Errorf("Failed to create PodGroup for pod %s: %v", klog.KObj(&pod), err)
```
Comment (Monokaix) [inline pkg/controllers/podgroup/pg_controller_handler.go:103]: same above.

File: pkg/controllers/podgroup/pg_controller_handler.go (hunk 2 above)
Comment (liuyuanchun11) [reply pkg/controllers/podgroup/pg_controller_handler.go:96]: The value must be a pointer. pod:= podList.Items[0], pod is not a pointer.

Comment (Monokaix) [reply pkg/controllers/podgroup/pg_controller_handler.go:96]: OK.

**Pull request merged**